### ISSUE TITLE
Uncaught exceptions in Promise: errors being thrown inside async callbacks

### ISSUE THREAD

**Issue by transcranial** [cornerstonejs/cornerstoneWADOImageLoader#180]
I'm encountering the following error, with no way of actually catching and handling it:

```
cornerstoneWADOImageLoader.min.js:2 Uncaught (in promise) Error: No decoder for transfer syntax 1.2.840.10008.1.2.4.55
    at o (cornerstoneWADOImageLoader.min.js:2)
    at u (cornerstoneWADOImageLoader.min.js:2)
    at eval (cornerstoneWADOImageLoader.min.js:2)
```

The error traces back to [this line](https://github.com/cornerstonejs/cornerstoneWADOImageLoader/blob/master/src/imageLoader/wadouri/loadImage.js#L29), originating from an error thrown [here](https://github.com/cornerstonejs/cornerstoneWADOImageLoader/blob/master/src/imageLoader/decodeImageFrame.js#L82). 

This error is being thrown inside an [asynchronous callback](https://github.com/cornerstonejs/cornerstoneWADOImageLoader/blob/master/src/imageLoader/wadouri/loadImage.js#L25-L56), so the wrapping Promise does not actually reject and an uncaught exception is generated.

This can be fixed either with a `.catch(...)` on `dataSetPromise`, or we can `Promise.reject(new Error(...))` instead of throwing the original error.

**Comment by swederik** [comment]
Should be fixed by #181.

Thanks for the report!

**Comment by transcranial** [comment]
Awesome, thanks!
